examples/java8/src/main/java/org/apache/beam/examples/complete/game/AggregatedTeamScore.java

```diff
@@ -0,0 +1,183 @@
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.beam.examples.common.ExampleUtils;
+import org.apache.beam.examples.complete.game.utils.WriteToBigQuery.FieldInfo;
+import org.apache.beam.examples.complete.game.utils.WriteWindowedToBigQuery;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.coders.VarIntCoder;
+import org.apache.beam.sdk.extensions.gcp.options.GcpOptions;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubIO;
+import org.apache.beam.sdk.options.Default;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.state.StateSpec;
+import org.apache.beam.sdk.state.StateSpecs;
+import org.apache.beam.sdk.state.ValueState;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.TypeDescriptor;
+import org.apache.beam.sdk.values.TypeDescriptors;
+import org.joda.time.Instant;
+
+/**
+ * This class is part of a series of pipelines that tell a story in a 'gaming' domain. Concepts
+ * include: stateful processing.
+ *
+ * <p>This pipeline processes an unbounded stream of 'game events'. It uses stateful processing to
+ * aggregate team scores per team and outputs team name and it's total score after every N points
+ * earned. In real world, N could be the total score a team should earn to level up. By default,
+ * this threshold (N) is set to 5000.
+ *
+ * <p>Stateful processing allows us to write pipelines that output based on a runtime state (when
+ * team reaches certain score, in every 100 game events etc) without time triggers. See
+ * https://beam.apache.org/blog/2017/02/13/stateful-processing.html for more information on using
+ * stateful processing.
+ *
+ * <p>Run {@code injector.Injector} to generate pubsub data for this pipeline.  The Injector
+ * documentation provides more detail on how to do this.
+ *
+ * <p>To execute this pipeline, specify the pipeline configuration like this:
+ * <pre>{@code
+ *   --project=YOUR_PROJECT_ID
+ *   --tempLocation=gs://YOUR_TEMP_DIRECTORY
+ *   --runner=YOUR_RUNNER
+ *   --dataset=YOUR-DATASET
+ *   --topic=projects/YOUR-PROJECT/topics/YOUR-TOPIC
+ * }
+ * </pre>
+ *
+ * <p>The BigQuery dataset you specify must already exist. The PubSub topic you specify should be
+ * the same topic to which the Injector is publishing.
+ */
+public class AggregatedTeamScore extends LeaderBoard {
+
+  /**
+   * Options supported by {@link AggregatedTeamScore}.
+   */
+  interface Options extends LeaderBoard.Options {
+
+    @Description("Numeric value of fixed window duration for team analysis, in minutes")
+    @Default.Integer(5000)
+    Integer getThreshold();
```

Comment:
an unqualified "Threshold" is too generic, and the description ("fixed windows") conflicts with an existing model concept. Something like "getMaximumBufferDuration(Minutes)"?

Reply:
The description message is copy/paste mistake. Updated.